climpred/metrics.py

```diff
@@ -41,16 +43,95 @@ def _get_norm_factor(comparison):
         KeyError: if comparison is not matching.
 
     """
-    comparison_name = comparison.__name__
-    if comparison_name in ['_m2e', '_e2c', '_e2r']:
+    if comparison.name in ['m2e', 'e2c', 'e2r']:
         fac = 1
-    elif comparison_name in ['_m2c', '_m2m', '_m2r']:
+    elif comparison.name in ['m2c', 'm2m', 'm2r']:
         fac = 2
     else:
         raise KeyError('specify comparison to get normalization factor.')
     return fac
 
 
+def _display_metric_metadata(self):
+    summary = '----- Metric metadata -----\n'
+    summary += f'Name: {self.name}\n'
+    summary += f'Longname: {self.longname}\n'
+    summary += f'Alias: {self.aliases}\n'
+    # positively oriented
+    if self.is_positive:
+        summary += 'Orientation: positive\n'
+    else:
+        summary += 'Orientation: negative\n'
+    # probabilistic or deterministic
+    if self.is_probabilistic:
+        summary += 'Kind: probabilistic\n'
+    else:
+        summary += 'Kind: deterministic\n'
+    summary += f'Power of units: {self.unit_power}\n'
+    summary += f'Minimum skill {self.min}\n'
+    summary += f'Maximum skill {self.max}\n'
+    summary += f'Perfect skill {self.perfect}\n'
+    summary += f'Strictly proper score: {self.proper}\n'
+    # doc
+    summary += f'Function: {self.function.__doc__}\n'
+    return summary
+
+
+class Metric:
+    'Master class for all metrics.'
+
+    def __init__(
+        self,
+        name,
+        function,
+        is_positive,
+        is_probabilistic,
+        longname=None,
+        aliases=None,
+        unit_power=1,
+        min=None,
+        max=None,
+        perfect=None,
+        proper=None,
+    ):
+        """Metric initialization.
+
+        Args:
+            name (str): name of metric.
+            function (function): metric function.
+            is_positive (bool): Is metric positively oriented? Higher metric
+             values means higher skill.
+            is_probabilistic (bool): Is metric probabilistic? `False` means
+             deterministic.
+            unit_power (float, int): Power of the input unit for unit of skill.
```

Comment:
I was sloppy with ordering. true.

unit power: mse has 2 because the unit [m] after mse is computed is [m**2], similar for pearson_r is 0. I will try to rephrase.